src/VisualStudio/Core/Def/Implementation/Experimentation/KeybindingResetDetector.cs

```diff
@@ -70,9 +72,13 @@ internal sealed class KeybindingResetDetector : ForegroundThreadAffinitizedObjec
         /// If false, ReSharper is either not installed, or has been disabled in the extension manager.
         /// If true, the ReSharper extension is enabled. ReSharper's internal status could be either suspended or enabled.
         /// </summary>
-        private bool _resharperExtensionEnabled = false;
-
+        private bool _resharperExtensionInstalled = false;
         private bool _infoBarOpen = false;
+        /// <summary>
+        /// True, when ReSharper is installed and running
+        /// False, when ReSharper is not installed, or when installed and suspended
+        /// </summary>
+        private bool _resharperRunning = false;
 
         [ImportingConstructor]
         [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
```

Comment:
in InitializeAsync, how about using UIKnownIssue to wait until solution is fully loaded?

http://source.roslyn.io/#Microsoft.VisualStudio.LanguageServices/Implementation/LanguageService/AbstractPackage.cs,13